codalab/worker/bundle_service_client.py

```diff
@@ -21,26 +21,29 @@ def wrapper(*args, **kwargs):
             except RestClientException as e:
                 raise BundleServiceException(message + ': ' + str(e), e.client_error)
             except urllib.error.HTTPError as e:
-                client_error = json.loads(e.read().decode())
-
-                if client_error['error'] == 'invalid_grant':
-                    raise BundleAuthException(
-                        message
-                        + ': '
-                        + http.client.responses[e.code]
-                        + ' - '
-                        + json.dumps(client_error),
-                        True,
-                    )
-                else:
-                    raise BundleServiceException(
-                        message
-                        + ': '
-                        + http.client.responses[e.code]
-                        + ' - '
-                        + json.dumps(client_error),
-                        e.code >= 400 and e.code < 500,
-                    )
+                try:
+                    client_error = json.loads(e.read().decode())
+
+                    if client_error['error'] == 'invalid_grant':
+                        raise BundleAuthException(
+                            message
+                            + ': '
+                            + http.client.responses[e.code]
+                            + ' - '
+                            + json.dumps(client_error),
+                            True,
+                        )
+                    else:
+                        raise BundleServiceException(
+                            message
+                            + ': '
+                            + http.client.responses[e.code]
+                            + ' - '
+                            + json.dumps(client_error),
+                            e.code >= 400 and e.code < 500,
+                        )
+                except json.decoder.JSONDecodeError:
+                    raise BundleServiceException(message + ': ' + str(e), False)
```

Comment:
Error 500 codes were causing JSON decode errors in the earlier code block, so this surfaces the actual error from the server instead.